torchgeo/datasets/dfc2022.py

```diff
@@ -0,0 +1,354 @@
+                out_shape=shape, out_dtype="float32", resampling=Resampling.bilinear
+            )
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load the target mask for a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the target mask
+        """
+        with rasterio.open(path) as f:
+            array: "np.typing.NDArray[np.int_]" = f.read(
+                indexes=1, out_dtype="int32", resampling=Resampling.bilinear
+            )
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            tensor = tensor.to(torch.long)  # type: ignore[attr-defined]
+            return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if checksum fails or the dataset is not downloaded
+        """
+        # Check if the files already exist
+        exists = []
+        for split_info in self.metadata.values():
+            exists.append(
+                os.path.exists(os.path.join(self.root, split_info["directory"]))
+            )
+
+        if all(exists):
+            return
+
+        # Check if .zip files already exists (if so then extract)
+        exists = []
+        for split_info in self.metadata.values():
+            filepath = os.path.join(self.root, split_info["filename"])
+            if os.path.isfile(filepath):
+                if self.checksum and not check_integrity(filepath, split_info["md5"]):
+                    raise RuntimeError("Dataset found, but corrupted.")
+                exists.append(True)
+                extract_archive(filepath)
+            else:
+                exists.append(False)
+
+        if all(exists):
+            return
+
+        # Check if the user requested to download the dataset
+        raise RuntimeError(
+            "Dataset not found in `root` directory, either specify a different"
+            + " `root` directory or manually download the dataset to this directory."
+        )
+
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        ncols = 2
+        image = sample["image"][:3]
+        image = image.to(torch.uint8)  # type: ignore[attr-defined]
+        image = image.permute(1, 2, 0).numpy()
+
+        dem = sample["image"][-1].numpy()
+        dem = percentile_normalization(dem, lower=0, upper=100, axis=(0, 1))
```

Comment:
This scales the DEM so that each image ranges from min to max? I wonder if we should add a colorbar to the side so that colors have meaning

Reply:
We don't do this other datasets (e.g. IDTReeS) so I'm leaning towards doing this for multiple datasets in a future PR.